frontend/public/components/environment.jsx

```diff
@@ -266,36 +272,57 @@ export const EnvironmentPage = connect(stateToProps)(
     }
 
     render() {
-      const {errorMessage, success, inProgress, currentEnvVars, stale, configMaps, secrets} = this.state;
+      const {errorMessage, success, inProgress, currentEnvVars, stale, configMaps, secrets, containerKey} = this.state;
       const {rawEnvData, readOnly, obj} = this.props;
       const isContainerArray = _.isArray(rawEnvData);
+      const containerDropdown = <ContainerDropdown
+        currentKey={containerKey}
+        containers={rawEnvData}
+        onChange={this.selectContainer} />;
 
       if (!configMaps || !currentEnvVars || !secrets) {
         return <LoadingBox />;
       }
-      const containerVars = currentEnvVars.map((envVar, i) => {
-        const keyString = isContainerArray ? rawEnvData[i].name : obj.metadata.name;
-        return <React.Fragment key={keyString}>
+      const envVar = currentEnvVars[containerKey];
+      const owners = _.get(obj.metadata, 'ownerReferences', [])
+        .map((o, i) => <ResourceLink key={i} kind={referenceForOwnerRef(o)} name={o.name} namespace={obj.metadata.namespace} title={o.uid} />);
+      const resourceName = _.get(obj.metadata, 'name', '');
+      const containerVars =
+        <React.Fragment>
+          { readOnly &&
+            <div className="row environment-alert">
+              <Alert className="col-md-11 col-xs-10" type="info">Environment variables for {resourceName} can only be edited through the resource {owners.length > 1 ? 'owners' : 'owner'}: <span className="environment-resource-link">
```

Comment:
I would change the wording here a little bit since you don't know for sure if you can edit the env variables for the owner either (for instance, in the pod -> rc -> dc case mentioned above). Maybe

```
Environment variables for {resourceName} were set from ...
```